indexer/routing.hpp

```diff
@@ -0,0 +1,198 @@
+#pragma once
+
+#include "coding/bit_streams.hpp"
+#include "coding/elias_coder.hpp"
+#include "coding/file_writer.hpp"
+#include "coding/reader.hpp"
+#include "coding/varint.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "base/assert.hpp"
+#include "base/bits.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/string.hpp"
+#include "std/vector.hpp"
+
+namespace routing
```

Comment:
I guess no. On the one hand Restriction is necessary in indexer, routing and generator. On the other hand logically it's a routing class. If I move routing::Restriction to routing indexer depends on router.